ang/civicase/ActivityActions--move-copy.js

```diff
@@ -0,0 +1,130 @@
+(function (angular, $, _) {
+  var module = angular.module('civicase');
+
+  module.service('MoveCopyActivityAction', MoveCopyActivityAction);
+
+  function MoveCopyActivityAction ($rootScope, crmApi, dialogService) {
+    var ts = CRM.ts('civicase');
+    var MOVE_COPY_ACTIVITY_LIMIT = 200;
+
+    /**
+     * Move/Copy activities
+     *
+     * @param {Array} activities
+     * @param {String} operation
+     */
+    this.moveCopyActivities = function (activities, operation) {
+      if (ifActivityLimitIsReached(activities)) {
+        return;
+      }
+
+      var activitiesCopy = _.cloneDeep(activities);
+      var title = operation[0].toUpperCase() + operation.slice(1) +
+        ((activities.length === 1)
+          ? ts(' %1Activity', {1: activitiesCopy[0].type ? activitiesCopy[0].type + ' ' : ''})
+          : ts(' %1 Activities', {1: activitiesCopy.length}));
+      var model = {
+        ts: ts,
+        case_id: activitiesCopy.length > 1 ? '' : activitiesCopy[0].case_id,
+        isSubjectVisible: activitiesCopy.length === 1,
+        subject: activitiesCopy.length > 1 ? '' : activitiesCopy[0].subject
+      };
+
+      dialogService.open('MoveCopyActCard', '~/civicase/ActivityMoveCopy.html', model, {
+        autoOpen: false,
+        height: 'auto',
+        width: '40%',
+        title: title,
+        buttons: [{
+          text: ts('Save'),
+          icons: {primary: 'fa-check'},
+          click: function () {
+            moveCopyConfirmationHandler.call(this, operation, activities, activitiesCopy, model);
+          }
+        }]
+      });
+    };
+
+    /**
+     * Handles the click event when the move/copy operation is confirmed
+     *
+     * @param {String} operation
+     * @param {Array} activities
+     * @param {Array} activitiesCopy
+     * @param {Object} model
+     */
+    function moveCopyConfirmationHandler (operation, activities, activitiesCopy, model) {
```

Comment:
`activitiesCopy` is not even required in this function, it was created so that when used in the modal's model, the original object does not get changed when user changes the value
```
var model = {
        ts: ts,
        case_id: activities.length > 1 ? '' : activitiesCopy[0].case_id,
        isSubjectVisible: activities.length === 1,
        subject: activities.length > 1 ? '' : activitiesCopy[0].subject
      };

      dialogService.open('MoveCopyActCard', '~/civicase/ActivityMoveCopy.html', model, {
```

Removed `activitiesCopy` from all the places where it is not required, please check.